src/components/ContactsPage.js

```diff
@@ -29,6 +32,23 @@ class ContactsPage extends React.Component {
       localStorage.clear();
     }
   }
+  
+  addContact = (firstName, lastName, image) => {
+    const { contacts } = this.state;
+    if (!(validateName(firstName) && validateName(lastName) && validateImageUrl(image))) {
+      return;
+    }
+    const newContact = {
+      id: generateId(),
+      firstName,
+      lastName,
+      image,
+      likes: 0,
+      isChecked: false
+    };
+    contacts.unshift(newContact);
```

Comment:
Again this is mutating the 'contacts' array. Use the spread operator instead:
`const updatedContacts = [newContact, ...contacts];`
or you can directly set that on setState:
`this.setState({ contacts: [ newContact, ...contacts ] }); `